src/components/generator/generatorFunction.js

```diff
@@ -67,20 +81,18 @@ export function generateRandomCombo(conditions, retries) {
  * Returns true if the current trick is a legal one in regard to the specified conditions. This does not check for
  * duplicates as it only looks at one trick and its predecessor in the combo.
  */
-function areLocalComboConditionsLegal(index, lastTrick, currentTrick, numberOfTricks, positionRules) {
-  const [allowConsecutiveTricks, allowTransitions, allowSimilarPositions, finishToFeet] = positionRules
-
-  const isConsecutiveTricksLegal = allowConsecutiveTricks || lastTrick.id !== currentTrick.id
+function areLocalComboConditionsLegal(index, lastTrick, currentTrick, conditions) {
+  const isConsecutiveTricksLegal = conditions.allowConsecutiveTricks || lastTrick.id !== currentTrick.id
 
   const isEndAndStartEqual = currentTrick.startPos === lastTrick.endPos
-  const isLegalSimilar = arePositionsSimilar(currentTrick.startPos, lastTrick.endPos) && allowSimilarPositions
-  const isTransitionLegal = isLegalSimilar || isEndAndStartEqual || allowTransitions
+  const isLegalSimilar = arePositionsSimilar(currentTrick.startPos, lastTrick.endPos) && conditions.allowSimilarPositions
+  const isTransitionLegal = isLegalSimilar || isEndAndStartEqual || conditions.allowTransitions
 
   if (!isConsecutiveTricksLegal || !isTransitionLegal) {
     return false
   }
 
-  if (index === numberOfTricks - 1 && finishToFeet) {
+  if (index === conditions.numberOfTricks - 1 && conditions.mustFinishToFeet) {
```

Comment:
nit: Personally, I would invert these conditions so you go from broad / generic conditions down to the more specific ones. But that is just a personal opinion. Feel free to disregard.

Reply:
Agreed, reads nicer the other way around. Changed it.